components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -1624,20 +1634,26 @@ bool ConversationHandler::IsContentAssociationPossible() {
 void ConversationHandler::BuildAssociatedContentInfo() {
   // Only modify associated content metadata here
   if (associated_content_delegate_) {
+    // Note: We don't create a new AssociatedContent object here unless one
+    // doesn't exist. If we generate one with a new UUID the deserializer
+    // breaks.
+    // TODO: Confirm with @petemill if I should try and fix this.
```

Comment:
This was an interesting one - presumably its because we were generating the AssociatedContent once and then changing the detail on it before, so the Id would never change.

I think a slightly better approach here would be to update the Id for the ArchiveContent when we do this, but I wasn't really sure of the implications.